homeassistant/core.py

```diff
@@ -2024,11 +2010,22 @@ async def async_store(self) -> None:
             "currency": self.currency,
         }
 
-        store: Store[dict[str, Any]] = Store(
-            self.hass,
-            CORE_STORAGE_VERSION,
-            CORE_STORAGE_KEY,
-            private=True,
-            atomic_writes=True,
-        )
+        store = self._ConfigStore(self.hass)
         await store.async_save(data)
+
+    # Circular dependency prevents us from generating the class at top level
+    # pylint: disable-next=import-outside-toplevel
+    from .helpers.storage import Store
+
+    class _ConfigStore(Store[dict[str, Any]]):
```

Comment:
Having it available as a class will help avoid discrepencies when adjusting the initialisation (for example when adding minor version and migration).